File: src/com/eagle/programmar/FSharp/FSharp_Type.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 11, 2022

package com.eagle.programmar.FSharp;

import com.eagle.programmar.FSharp.Terminals.FSharp_Keyword;
import com.eagle.programmar.FSharp.Terminals.FSharp_KeywordChoice;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.interfaces.AbstractType;
import com.eagle.transform.EagleGenerator.TypeEnum;

public class FSharp_Type extends TokenSequence implements AbstractType
{
	public @S(10) FSharp_KeywordChoice PRIMITIVE = new FSharp_KeywordChoice("bool", "int", "string");
	public @S(20) @OPT FSharp_Keyword ARRAY = new FSharp_Keyword("array");

	public TypeEnum findType()
	{
		if (ARRAY != null && ARRAY.isPresent())
		{
			switch (PRIMITIVE.getValue())
			{
			case "string":
				return TypeEnum.ARRAY;
			default:
				return TypeEnum.VOID;
			}
		}
		
		switch (PRIMITIVE.getValue())
		{
		case "bool":
			return TypeEnum.BOOLEAN;
		case "int":
			return TypeEnum.INTEGER;
		case "string":
			return TypeEnum.STRING;
		default:
			return TypeEnum.VOID;
		}
	}
}

File: csrc/com/eagle/programmar/FSharp/FSharp_Type.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Jul 11, 2022

namespace com.eagle.programmar.FSharp
{
	using FSharp_KeywordChoice = com.eagle.programmar.FSharp.Terminals.FSharp_KeywordChoice;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using AbstractType = com.eagle.tokens.interfaces.AbstractType;
	using TypeEnum = com.eagle.transform.EagleGenerator.TypeEnum;

	public class FSharp_Type : TokenChooser, AbstractType
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE FSharp_KeywordChoice XXTYPES = new com.eagle.programmar.FSharp.Terminals.FSharp_KeywordChoice("bool", "int", "string");
		public FSharp_KeywordChoice XXTYPES = new FSharp_KeywordChoice("bool", "int", "string");

		public static TypeEnum findType(FSharp_Type type)
		{
			FSharp_KeywordChoice typeName = (FSharp_KeywordChoice) type.getWhich();
			switch (typeName.getValue())
			{
			case "bool":
				return TypeEnum.BOOLEAN;
			case "int":
				return TypeEnum.INTEGER;
			case "string":
				return TypeEnum.STRING;
			default:
				return TypeEnum.VOID;
			}
		}
	}

}
